pkg/pureutils/purefa_rest2x.go

```diff
@@ -20,17 +19,21 @@ func PureCreateClientAndConnectRest226(faMgmtEndpoint string, apiToken string) (
 }
 
 // ListAllVolumesFromFA returns list of all Available Volumes present in FA (Function should be used with RestAPI 2.x)
-func ListAllVolumesFromFA(faClient *fa.Client) ([]fa.Volumes, error) {
-	volumes, err := faClient.Volumes.ListAllAvailableVolumes(nil, nil)
+func ListAllVolumesFromFA(faClient *flasharray.Client) ([]flasharray.VolResponse, error) {
+	params := make(map[string]string)
+	params["destroyed"] = "false"
+	volumes, err := faClient.Volumes.ListAllAvailableVolumes(params, nil)
 	if err != nil {
 		return nil, err
 	}
 	return volumes, nil
 }
 
 // ListAllDestroyedVolumesFromFA Returns list of all Destroyed FA Volumes (Function should be used with RestAPI 2.x)
-func ListAllDestroyedVolumesFromFA(faClient *fa.Client) ([]fa.Volumes, error) {
-	volumes, err := faClient.Volumes.ListAllDestroyedVolumes(nil, nil)
+func ListAllDestroyedVolumesFromFA(faClient *flasharray.Client) ([]flasharray.VolResponse, error) {
+	params := make(map[string]string)
+	params["destroyed"] = "true"
```

Comment:
is params being used some where in this function?

Reply:
yes, basically params are the query params that can be provided to the HTTP Request 



params map[string]string, data interface{}